core/src/components/route/route.tsx

```diff
@@ -28,6 +30,22 @@ export class Route implements ComponentInterface {
    */
   @Prop() componentProps?: {[key: string]: any};
 
+  /**
+   * A navigation hook that is fired when the route tries to leave.
+   * Returning `true` allows the navigation to proceed, while returning
+   * `false` causes it to be cancelled. Returning a string path causes
```

Comment:
This is needs to be updated to how the NavigationHookCallback object is, where it's no longer a string but an object